src/chartsFactory.jsx

```diff
@@ -1,4 +1,5 @@
 var React = require('react');
+var win = typeof global === 'undefined' ? window : global;
```

Comment:
So that it performs the check dynamically at runtime with every execution rather than once upon instantiation? Unless you think it's somehow possible for the global / window object to change within the same runtime, I would vote no ... but it is your project, I shall do what you wish. :)